mono/metadata/object.c

```diff
@@ -7841,9 +7841,10 @@ mono_get_eh_callbacks (void)
 void
 mono_raise_exception (MonoException *ex) 
 {
-	MONO_ENTER_GC_UNSAFE;
+	/* raise_exception doesn't return, so the transition to GC Unsafe is unbalanced */
+	MONO_STACKDATA (stackdata);
+	mono_threads_enter_gc_unsafe_region_unbalanced_with_info (mono_thread_info_current (), &stackdata);
```

Comment:
I tried something similar, using `mono_threads_enter_gc_unsafe_region_unbalanced ()` which basically expands to `mono_threads_enter_gc_unsafe_region_unbalanced_with_info (mono_thread_info_current_unchecked (), &stackdata)`.

so it's `mono_thread_info_current ()` vs. `mono_thread_info_current_unchecked ()`. Any reason why we don't use `mono_thread_info_current ()` in `mono_threads_enter_gc_unsafe_region_unbalanced ()`?

Reply:
ISTR `mono_threads_enter_gc_unsafe_region_unbalanced` (with the `_unchecked` thread info) can be called by products on detached threads in some circumstance.

But the real reason is because there's a dozen of these functions and I always forget about them.